### PR TITLE
WIP: testing/runc: new aport

### PR DESCRIPTION
**Pull request by deedubs**
CLI tool for spawning and running containers
according to the OCI specification

### CODE REVIEW

File: testing/runc/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Maintainer: Dan Williams <dan@ma.ssive.co>
+pkgname=runc
+pkgver=1.0.0_rc5
+pkgrel=0
+pkgdesc="CLI tool for spawning and running containers according to the OCI specification"
+url="https://github.com/opencontainers/runc"
+arch="all"
+license="Apache-2.0"
+depends=""
```
Comment (Ikke) [inline testing/runc/APKBUILD]: Empty variables should be left out.

File: testing/runc/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+makedepends="go bash libseccomp-dev linux-headers"
+install=""
+subpackages=""
+source="https://github.com/opencontainers/runc/archive/v${pkgver/_/-}.tar.gz
+	"
```
Comment (Ikke) [inline testing/runc/APKBUILD]: Looks better when you move this quote to the previous line.

File: testing/runc/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+builddir="$srcdir/$pkgname-${pkgver/_/-}"
+
+build() {
+	export AUTO_GOPATH=1
```
Comment (Ikke) [inline testing/runc/APKBUILD]: This seems to be a variable related to building Docker, nothing Go uses itself as far as I can tell.

File: testing/runc/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	cd $builddir
+	mkdir -p src/github.com/opencontainers/
+	ln -s "$builddir" src/github.com/opencontainers/runc
+	cd src/github.com/opencontainers/runc
+	GOPATH="$PWD" make COMMIT="$_runc_ver"
+}
+
+check() {
+  cd $builddir
+  ./runc --help > /dev/null
```
Comment (Ikke) [inline testing/runc/APKBUILD]: runc has a test suite. You should run that rather then just invoking --help.

File: testing/runc/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+builddir="$srcdir/$pkgname-${pkgver/_/-}"
+
+build() {
+	export AUTO_GOPATH=1
+
+	cd $builddir
+	mkdir -p src/github.com/opencontainers/
+	ln -s "$builddir" src/github.com/opencontainers/runc
+	cd src/github.com/opencontainers/runc
+	GOPATH="$PWD" make COMMIT="$_runc_ver"
```
Comment (jirutka) [inline testing/runc/APKBUILD]: `$_runc_ver` is not defined anywhere.

Comment (deedubs) [reply testing/runc/APKBUILD]: Ahh okay good to know!

**Comment by deedubs** [comment]
@jirutka @Ikke Thank you for taking the time to review this.  I'll apply your fixes shortly.

**Comment by BernhardGruen** [comment]
@deedubs To execute the tests you need to start a docker environment.
I already tried this for the _community/docker_ build itself. Unfortunately without success so far.
Reason: was not able to start dockerd inside the build. such that the tests can access the daemon.

**Comment by andypost** [comment]
This already commited e3a6a34509c08e45625c940842bdde6bf55af6bc

**Pull request closed without merging**